File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] suf = new int[nums.length];
        for(int i = nums.length - 1, q = 1; i >= 0; i--){
            suf[i] = q;
            q = q * nums[i];
        }

        int[] res = new int[nums.length];
        for(int i = 0, p = 1; i < res.length; i++){
            res[i] = p * suf[i];
            p = p * nums[i];
        }

        return res;
    }
}  

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        # [ , ]

        p, q = 0, len(nums) - 1

        while p < q:
            mid = (p + q) // 2

            if nums[mid] > nums[q]:
                p = mid + 1
            else:
                q = mid

        return nums[q]
